pkg/controller/capacity/capacity_controller.go

```diff
@@ -168,116 +172,122 @@ func (c *Controller) processCapacityTargetOnCluster(
-	)
+	if replicas.AchievedDesiredReplicaPercentage(spec.TotalReplicaCount, availableReplicas, spec.Percent) {
+		readyCond = capacityutil.NewClusterCapacityCondition(
+			shipper.ClusterConditionTypeReady,
+			corev1.ConditionTrue,
+			"",
+			"",
+		)
 
-	report, err := c.getReport(targetDeployment, status)
-	if err != nil {
-		return err
+		return nil
 	}
-	status.Reports = []shipper.ClusterCapacityReport{*report}
 
-	podCount, sadPodCount, sadPods, err := c.getSadPodsForDeploymentOnCluster(informerFactory, targetDeployment)
-	if err != nil {
-		cond := capacityutil.NewClusterCapacityCondition(
-			shipper.ClusterConditionTypeOperational,
-			corev1.ConditionFalse,
-			ServerError,
-			err.Error(),
-		)
-		diff.Append(capacityutil.SetClusterCapacityCondition(status, *cond))
-		return err
-	}
+	// Not all pods are availble, so we know for sure this cluster isn't
+	// ready. We try to give the user a good error message by checking
+	// whether the pods were created, but aren't ready (aka a sad pod) of
+	// if they were not created yet (possibly because of quota).
+	var msg string
+	sadPods = c.getSadPods(pods)
+	if l := len(sadPods); l > 0 {
+		if l > SadPodLimit {
+			sadPods = sadPods[:SadPodLimit]
+		}
 
-	if targetDeployment.Spec.Replicas == nil || int(*targetDeployment.Spec.Replicas) != podCount {
-		cond := capacityutil.NewClusterCapacityCondition(
-			shipper.ClusterConditionTypeReady,
-			corev1.ConditionFalse,
-			WrongPodCount,
-			fmt.Sprintf("expected %d replicas but have %d", *targetDeployment.Spec.Replicas, podCount),
+		msg = fmt.Sprintf(
+			"%d out of %d pods are sad. this might require intervention, check SadPods in this object for more information",
+			l, desiredReplicas,
 		)
-		diff.Append(capacityutil.SetClusterCapacityCondition(status, *cond))
-		return err
-	}
-
-	if sadPodCount > 0 {
-		cond := capacityutil.NewClusterCapacityCondition(
-			shipper.ClusterConditionTypeReady,
-			corev1.ConditionFalse,
-			PodsNotReady,
-			fmt.Sprintf("there are %d sad pods", sadPodCount),
+	} else {
+		msg = fmt.Sprintf(
+			"%d out of %d pods were not created yet. this might require intervention, check the Deployment for more information",
+			desiredReplicas-availableReplicas, desiredReplicas,
 		)
-		diff.Append(capacityutil.SetClusterCapacityCondition(status, *cond))
 	}
 
-	status.SadPods = sadPods
-
-	if sadPodCount == 0 {
-		condReady := capacityutil.NewClusterCapacityCondition(
-			shipper.ClusterConditionTypeReady,
-			corev1.ConditionTrue,
-			"",
-			"")
-		diff.Append(capacityutil.SetClusterCapacityCondition(status, *condReady))
-
-		condOperational := capacityutil.NewClusterCapacityCondition(
-			shipper.ClusterConditionTypeOperational,
-			corev1.ConditionTrue,
-			"",
-			"")
-		diff.Append(capacityutil.SetClusterCapacityCondition(status, *condOperational))
-	}
+	readyCond = capacityutil.NewClusterCapacityCondition(
+		shipper.ClusterConditionTypeReady,
+		corev1.ConditionFalse,
+		PodsNotReady,
+		msg,
+	)
```

Comment:
Not sure I red this correctly: no matter if there are some or no sad pods, ClusterConditionTypeReady is always being set to false here, right?

Reply:
Correct. This happens, for instance, when you exceed quotas: you can't create pods, so they won't be sad, but you still don't have enough pods.